src/app/shared/components/error-messages/error-messages.component.html

```diff
@@ -0,0 +1,11 @@
+<ng-container *ngFor="let validation of field?.validations">
+  <div
+    class="text-danger"
+    *ngIf="
```

Comment:
i think it's better to move such complex logic to the separate method, so it could be tested

Reply:
your right, done